apis/apps/v1alpha1/cluster_types.go

```diff
@@ -394,6 +394,19 @@ type UserResourceRefs struct {
 	ConfigMapRefs []ConfigMapRef `json:"configMapRefs,omitempty"`
 }
 
+// InstanceUpdateStrategy indicates the strategy that the InstanceSet
+// controller will use to perform updates. It includes any additional parameters
+// necessary to perform the update for the indicated strategy.
+type InstanceUpdateStrategy struct {
+	// Type indicates the type of the InstanceUpdateStrategy.
+	// Default is RollingUpdate.
+	// +optional
+	Type workloads.InstanceSetUpdateStrategyType `json:"type,omitempty"`
```

Comment:
It is not recommended to use types from apis/workloads directly in apis/apps, for the following reasons:

1. This would require users of apis/apps to be aware of and understand the API structure of apis/workloads.
2. Any changes made to apis/workloads would propagate to apis/apps, potentially causing unintended side effects or breaking changes.

Reply:
You can define a struct in both apis/apps and apis/workload, and then copy it over in the controller. @YTGhost

Reply:
> You can define a struct in both apis/apps and apis/workload, and then copy it over in the controller. @YTGhost

Okay, it looks reasonable.